test/controllers/update/update.managing.officer.type.controller.spec.ts

```diff
@@ -0,0 +1,137 @@
+      mockGetApplicationData.mockReturnValueOnce({
+        ...APPLICATION_DATA_MOCK,
+        [BeneficialOwnerStatementKey]: BeneficialOwnersStatementType.NONE_IDENTIFIED
+      });
+      const resp = await request(app).get(config.UPDATE_MANAGING_OFFICER_TYPE_URL);
+
+      expect(resp.status).toEqual(200);
+      expect(resp.text).toContain(BENEFICIAL_OWNER_TYPE_PAGE_HEADING_NONE_IDENTIFIED);
+      expect(resp.text).toContain(BENEFICIAL_OWNER_TYPE_PAGE_CORPORATE_MO);
+      expect(resp.text).toContain(BENEFICIAL_OWNER_TYPE_PAGE_INDIVIDUAL_MO);
+      expect(resp.text).not.toContain(BENEFICIAL_OWNER_TYPE_PAGE_HEADING_ALL_IDENTIFIED_ALL_DETAILS);
+      expect(resp.text).not.toContain(BENEFICIAL_OWNER_TYPE_PAGE_INDIVIDUAL_BO);
+      expect(resp.text).not.toContain(BENEFICIAL_OWNER_TYPE_PAGE_CORPORATE_BO);
+      expect(resp.text).not.toContain(BENEFICIAL_OWNER_TYPE_PAGE_GOVERNMENT_BO);
+    });
+
+    test(`renders the ${config.UPDATE_MANAGING_OFFICER_TYPE_PAGE} for managing officers with just the MOs options`, async () => {
+      mockGetApplicationData.mockReturnValueOnce({ [BeneficialOwnerStatementKey]: BeneficialOwnersStatementType.NONE_IDENTIFIED });
+      const resp = await request(app).get(config.UPDATE_MANAGING_OFFICER_TYPE_URL);
+
+      expect(resp.status).toEqual(200);
+      expect(resp.text).toContain(BENEFICIAL_OWNER_TYPE_PAGE_HEADING_NONE_IDENTIFIED);
+      expect(resp.text).toContain(BENEFICIAL_OWNER_TYPE_LEGEND_TEXT_NONE_IDENTIFIED);
+      expect(resp.text).toContain(BENEFICIAL_OWNER_TYPE_ADD_BUTTON_NONE_IDENTIFIED);
+    });
+
+    test("catch error when rendering the page", async () => {
+      mockGetApplicationData.mockImplementationOnce(() => { throw ERROR; });
+      const resp = await request(app).get(config.UPDATE_MANAGING_OFFICER_TYPE_URL);
+
+      expect(resp.status).toEqual(500);
+      expect(resp.text).toContain(SERVICE_UNAVAILABLE);
+    });
+  });
+
+  describe("POST tests", () => {
+    test(`redirects to the ${config.UPDATE_MANAGING_OFFICER_INDIVIDUAL_PAGE} page`, async () => {
+      const resp = await request(app)
+        .post(config.UPDATE_MANAGING_OFFICER_TYPE_URL)
+        .send({ [BeneficialOwnerTypeKey]: ManagingOfficerTypeChoice.individual });
+
+      expect(resp.status).toEqual(302);
+      expect(resp.header.location).toEqual(config.UPDATE_MANAGING_OFFICER_INDIVIDUAL_URL);
+    });
+
+    test(`redirects to the ${config.UPDATE_MANAGING_OFFICER_CORPORATE_PAGE} page`, async () => {
+      const resp = await request(app)
+        .post(config.UPDATE_MANAGING_OFFICER_TYPE_URL)
+        .send({ [BeneficialOwnerTypeKey]: ManagingOfficerTypeChoice.corporate });
+
+      expect(resp.status).toEqual(302);
+      expect(resp.header.location).toEqual(config.UPDATE_MANAGING_OFFICER_CORPORATE_URL);
+    });
+
+    test(`renders the current page with error message when ${BeneficialOwnersStatementType.NONE_IDENTIFIED} has been selected `, async () => {
```

Comment:
This test is out of place / BO test. Just delete I think.

Reply:
Test deleted